### PR TITLE
Update py-neurodamus and NEURON version

### PR DESCRIPTION
**Pull request by iomaganaris**
- Update `py-neurodamus` version with improvements in model clean up
- Use new `common` tag with improvements in synapse freeing
- Update `neuron` version with updates on version `8.2.1` related to model cleanup

### CODE REVIEW

File: bluebrain/repo-bluebrain/packages/py-neurodamus/package.py [hunk 1]
```diff
@@ -13,6 +13,7 @@ class PyNeurodamus(PythonPackage):
     git = "ssh://git@bbpgitlab.epfl.ch/hpc/sim/neurodamus-py.git"
 
     version('develop', branch='main', submodules=True)
+    version('2.12.9',  branch='magkanar/clear_model', submodules=True)
```
Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/py-neurodamus/package.py]: No branches for fixed versions, please. Use a commit if you can't do a tag.

Comment (iomaganaris) [reply bluebrain/repo-bluebrain/packages/py-neurodamus/package.py]: Yes, this is just temporary to test everything together. I will replace this when the PR is ready for review. Should I also clear the PR build directory if I replace this with the tag and keep the same version?

Comment (matz-e) [reply bluebrain/repo-bluebrain/packages/py-neurodamus/package.py]: Yes, please. In fact, for every proper rebuild you'd have to clear the PR build directory (while testing)

Comment (iomaganaris) [reply bluebrain/repo-bluebrain/packages/py-neurodamus/package.py]: I've now updated the new py-neurodamus version with the proper tag. Thanks for your help!

**Review by jorblancoa — APPROVED** [review]

File: bluebrain/repo-patches/packages/neuron/package.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ class Neuron(CMakePackage):
     patch("patch-v800-cmake-nvhpc.patch", when="@8.0.0%nvhpc^cmake@3.20:")
 
     version("develop", branch="master")
+    version("8.2.2a", commit="eb19ae0")
```
Comment (pramodk) [inline bluebrain/repo-patches/packages/neuron/package.py:34]: Just copying the slack discussing here:  when we will have the actual 8.2 release then this version name will be problematic:

```python
>>> Version("8.2.2") > Version("8.2.2a")
False
```
i.e. such version naming doesn't work well with Spack. Some alternatives are like:

```python
>>> Version("9.b1") > Version("9.a1")
True
>>> Version("9.a2") > Version("9.a1")
True
>>> Version("9.0") > Version("9.a1")
True
```

**Pull request merged**